aml-development/frontend
# feat(cache): proof of concept of caching actions using redux

**Description**
Create the ability to cache actions/service calls to prevent unnecessary API calls to the backend

**Acceptance Criteria**
1. For specific endpoints (ie getting the current user's profile), result data is cached.
2. The cache is invalidated when the data becomes stale (ie profile is updated).

**How To Test**
Pull this branch, `npm install`.  Most of the caching can be seen via the browser's dev console.

1.  Go to the profile page.  API endpoint to fetch profile is triggered.
2.  Go back to home page or to a listing page.  Go back to profile page.  Profile displays, but endpoint is no longer triggered.
3.  Click the edit profile button.  See that the cache was cleared in the logs.
4.  Go back to profile.  API endpoint to fetch profile is triggered.

NOTE: the caching occurs in the Redux state and is not persisted to local/session storage or a cookie.  This means that if you manually refresh the page, the cached data is gone.  If this is not desired, we can look at a framework that persists/hydrates redux stores to/from storage, such as https://github.com/rt2zz/redux-persist
